executor/explainfor_test.go

```diff
@@ -486,3 +487,35 @@ func (s *testPrepareSerialSuite) TestPointGetUserVarPlanCache(c *C) {
 		"2 4 2 2",
 	))
 }
+
+func (s *testPrepareSerialSuite) TestExpressionIndexPreparePlanCache(c *C) {
+	tk := testkit.NewTestKitWithInit(c, s.store)
+
+	orgEnable := core.PreparedPlanCacheEnabled()
+	defer func() {
+		core.SetPreparedPlanCache(orgEnable)
+	}()
+	core.SetPreparedPlanCache(true)
+	config.UpdateGlobal(func(conf *config.Config) {
+		conf.Experimental.AllowsExpressionIndex = true
+	})
+
+	var err error
+	tk.Se, err = session.CreateSession4TestWithOpt(s.store, &session.Opt{
+		PreparedPlanCache: kvcache.NewSimpleLRUCache(100, 0.1, math.MaxUint64),
+	})
+	c.Assert(err, IsNil)
+
+	tk.MustExec("use test")
+	tk.MustExec("set @@tidb_enable_collect_execution_info=0;")
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t(a int, b int, key ((a+b)));")
+	tk.MustExec("prepare stmt from 'select * from t where a+b = ?'")
+	tk.MustExec("set @a = 123")
+	tk.MustExec("execute stmt using @a")
+
+	tkProcess := tk.Se.ShowProcess()
+	ps := []*util.ProcessInfo{tkProcess}
+	tk.Se.SetSessionManager(&mockSessionManager1{PS: ps})
+	c.Assert(tk.MustUseIndex("for connection "+strconv.FormatUint(tkProcess.ID, 10), "expression_index"), IsTrue)
+}
```

Comment:
Please complete this test for the reusing of the cached plan, i.e, change the value of `@a`, execute the stmt again, and check the range used in the cached plan is correct.